### PR TITLE
Add bullet to ports (using ammo.js)

### PR DESCRIPTION
**Pull request by nadult**
Hi, Can you please pull this?
I didn't change anything in ammo.js, only added a release named 'version_1' to be consistent with other ports.
Btw. before pulling, link to ammo.js repo in ports/bullet.py has to be changed (now it points to my repo).

### CODE REVIEW

File: tools/ports/bullet.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      return os.path.join(dest_path, 'libbullet.bc')
+    return [shared.Cache.get('bullet', create)]
+  else:
+    return []
+
+
+def process_args(ports, args, settings, shared):
+  if settings.USE_BULLET == 1:
+    get(ports, settings, shared)
+    args += ['-Xclang', '-isystem' + os.path.join(shared.Cache.get_path('ports-builds'), 'bullet', 'bullet', 'src')]
```
Comment (kripken) [inline tools/ports/bullet.py]: Does this need to be `isystem`? Couldn't it be a normal `-I`?

**Comment by kripken** [comment]
I created https://github.com/emscripten-ports/Bullet and added you as an owner, because in general in ports that's how we've done things. However, perhaps using the existing ammo.js repo makes sense? I'm open to being convinced either way.

At first thought, it seems like for consistency, a repo in ports seems best, but on the other hand, we already have ammo.js.

Another option might be to move ammo.js to a ports repo, perhaps.

File: tools/ports/bullet.py (hunk 1 above)
Comment (nadult) [inline tools/ports/bullet.py]: Both will work, but other libraries use -isystem, that's why I used it here.  Advantage of -isystem is that warnings from bullet headers will be suppressed (unless user explicitly enables them).

**Comment by nadult** [comment]
IMO having two repositories is the best solution. Each repository serves a different purpose:
- in ports when someone wants to quickly use it to port C++ code
- as ammo.js when someone uses it on the Web through javascript layer

File: tools/ports/bullet.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  return args
+
+def show():
+  return 'bullet (zlib license)'
```
Comment (kripken) [inline tools/ports/bullet.py]: This should be `'bullet (USE_BULLET=1; zlib license)'`, see the other ports on the incoming branch. (In the future please make pulls directly to there, although for this one, I don't see a merge conflict so it's ok.)

**Comment by kripken** [comment]
The test fails over here,

````
$ ./tests/runner.py other.test_bullet
test_bullet (test_other.other) ... WARNING  root: retrieving port: bullet from https://github.com/emscripten-ports/bullet/archive/version_1.zip
WARNING  root: unpacking port: bullet
WARNING  root: generating system library: bullet.bc...
WARNING  root: building port: bullet
running aclocal
running libtool
autogen.sh: 18: autogen.sh: libtoolize: not found
libtool not found. trying glibtool.
autogen.sh: 29: autogen.sh: glibtoolize: not found
An error occured, autogen.sh stopping.

--------------------------------------------------
Building ammo.js, build type: ['-O3', '--llvm-lto', '1', '-s', 'NO_EXIT_RUNTIME=1', '-s', 'AGGRESSIVE_VARIABLE_ELIMINATION=1', '-s', 'NO_DYNAMIC_EXECUTION=1', '--memory-init-file', '0', '-s', 'NO_FILESYSTEM=1', '-s', 'NO_BROWSER=1', '-s', 'TOTAL_MEMORY=67108864', '-s', 'EXPORT_NAME="AmmoLib"', '-s', 'MODULARIZE=1']
--------------------------------------------------


==========================
Stage 1: Generate bindings
==========================


=======================
Stage 2: Build bindings
=======================


===================================================================
Stage 3: Configure (if this fails, run autogen.sh in bullet/ first)
===================================================================

ERROR    root: Exception thrown when invoking Popen in configure with args: "../configure --disable-demos --disable-dependency-tracking"!
Traceback (most recent call last):
  File "make.py", line 108, in <module>
    emscripten.Building.configure(['../configure', '--disable-demos','--disable-dependency-tracking'])
  File "/media/alon/2f9a30d7-6124-42d9-87c5-3c80cb70ec54/home/alon/Dev/emscripten/tools/shared.py", line 1098, in configure
    process = Popen(args, stdout=None if EM_BUILD_VERBOSE_LEVEL >= 2 else stdout, stderr=None if EM_BUILD_VERBOSE_LEVEL >= 1 else stderr, env=env)
  File "/usr/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1327, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory
````

In general, depending on `configure` is something we've ended up avoiding in other ports, because of similar problems. We generally run configure once and check in the output files, see `zlib` for example.

**Comment by nadult** [comment]
I've created a new pull request with fixed issues and more libraries:
https://github.com/kripken/emscripten/pull/3646

**Pull request closed without merging**